api/lib/mno_enterprise/intercom_events_listener.rb

```diff
@@ -0,0 +1,79 @@
+require 'intercom'
+
+module MnoEnterprise
+
+  class IntercomEventsListener
+
+    attr_accessor :intercom
+
+    def initialize
+      if MnoEnterprise.intercom_api_key && MnoEnterprise.intercom_api_secret
+        self.intercom = ::Intercom::Client.new(app_id: MnoEnterprise.intercom_api_key, api_key: MnoEnterprise.intercom_api_secret)
+      else
+        self.intercom = nil
+      end
+
+    end
+
+    def info(key, current_user_id, description, metadata, object)
+      return unless self.intercom
+      u = User.find(current_user_id)
+      begin
+        begin
+          intercom.users.find(:user_id => current_user_id)
+        rescue Intercom::ResourceNotFound
+          self.update_intercom_user(u)
+        end
+        case key
+          when 'user_update'
+            self.update_intercom_user(u)
+            return
+          when 'app_destroy'
+            event_name = 'deleted-app-' + object.app.nid
+          when 'app_add'
+            event_name = 'added-app-' + object.app.nid
+          else
+            event_name = key.gsub!(/_/, '-')
```

Comment:
I can show off a new trick I learned! I you don't use the regex capture groups in the replacement or if like here you're just doing a string substitution, `#tr` is order of magnitude faster than `#gsub`

See [this benchmark](https://github.com/JuanitoFatas/fast-ruby/blob/master/code/string/gsub-vs-tr.rb):

```
Warming up --------------------------------------
         String#gsub    49.395k i/100ms
           String#tr   134.293k i/100ms
Calculating -------------------------------------
         String#gsub    542.159k (±23.6%) i/s -      2.519M in   5.001670s
           String#tr      2.346M (± 5.8%) i/s -     11.683M in   5.001330s

Comparison:
           String#tr:  2345516.8 i/s
         String#gsub:   542158.9 i/s - 4.33x  slower
```